reactor-pool/src/main/java/reactor/pool/SimpleDequePool.java

```diff
@@ -451,7 +446,7 @@ else if (sig.isOnError()) {
 							final int mergeConcurrency = Math.min(poolConfig.allocationStrategy().warmupParallelism(), toWarmup + 1);
 							Flux.range(1, toWarmup)
 									.map(i -> warmupMono(i, toWarmup, startWarmupIteration, allocator).doOnSuccess(__ -> drain()))
-									.startWith(primary.doOnSuccess(__ -> drain()).then())
+									.startWith(primary.doOnSuccess(__ -> drain()).then().onErrorComplete())
```

Comment:
```suggestion
									.startWith(primary.doOnSuccess(__ -> drain()).onErrorComplete().then())
```

although the patch seems correct, what do you think if we append `onErrorComplete` just after the `primary.doOnSuccess(...)`, it is just a matter of taste. I think it's the same, I just tend to prefer doing that in terms of clarity ?

@chemicL , do you agree with this patch ? and are you ok to move the onErrorComplete before `.then()`(just for readability reasons) ?